Automattic/newspack-theme
# Feature: Add option to display 'updated' date

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds an option to the customizer to display each post's last updated date, if it has been updated more than a full day after its published.

When paired with the 'time ago' date format, it will automatically override the 'cut off' time for the time ago format, and switch it to 1 day. This is to serve two purposes: 1. Make sure you don't end up with a somewhat unclear date, like "November 20th Updated 2 days ago", and 2. to avoid having both use the time ago format without too many code acrobatics (5 days ago Update 2 days ago").

**Questions** 

* Do we need some kind of override for changes that happen during migration/launch? I noticed this in the CalMatters code I was working from.
* Right now, the updated dates only display on single posts; would it make sense to add them to other locations, too, like the archives, search, or blocks (which would, in part, need a separate PR at least to add the Updated string)? 
* Does the 'Time ago' override behaviour make sense, or would it be better to honour their settings and let them decide if it seems weird or not?

See #598

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Start with a mix of posts that have been published recently, updated recently, and published and updated again in the same 24 hour period. 
3. Navigate to Customizer > Template Settings > Post Settings and check the 'Show "Last updated" date' checkbox.
4. Confirm that on posts that have been published and updated more than 24 hours later, a second 'Updated' date is displaying:

![image](https://user-images.githubusercontent.com/177561/100296881-2af55400-2f42-11eb-9676-678040e610bc.png)

5. Confirm that on posts that have been updated less than 24 hours after publishing, the updated date does not show up.
6. Navigate back to Customizer > Template Settings > Post settings, and check 'Use "time ago" date format'; note that the 'Cut off' field is greyed out.
7. Try toggling on and off the 'Show "last updated" date' to make sure the cut off field is enabled/disabled. Leave both options checked and click 'Publish'.
8. Confirm that the time-ago date is only applied to the updated date, even when the normal 'cut-off' would apply it to both dates:

![image](https://user-images.githubusercontent.com/177561/100297071-b66ee500-2f42-11eb-8641-a5a0906be18c.png)

9. Confirm that posts that don't display the updated date (either because they haven't been updated, or that were updated in the 24 hours after publishing), don't show the updated date, and show the 'time ago' format:

![image](https://user-images.githubusercontent.com/177561/100297217-077ed900-2f43-11eb-99d8-7951c4ba6658.png)

### Other information:
